### PR TITLE
[Pulsar-Perf] Allow configuring `metadataStoreUrl` in `pulsar-perf managed-ledger`

### PR DESCRIPTION
**Pull request by RobertIndie**
Master Issue: #13760

### Motivation

As per #13760, `zookeeperServers` need to be renamed to `metadataStoreUrl`.

### Modifications

* Add `metadataStoreUrl` parameter to ManagedLedgerWriter of the pulsar perf.

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

- [x] `doc-required`

### CODE REVIEW

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java [hunk 1]
```diff
@@ -98,10 +98,14 @@
                 description = "Number of threads writing", validateWith = PositiveNumberParameterValidator.class)
         public int numThreads = 1;
 
-        @Parameter(names = { "-zk", "--zookeeperServers" },
-                description = "ZooKeeper connection string", required = true)
+        @Parameter(names = {"-zk", "--zookeeperServers"},
+                description = "ZooKeeper connection string")
```
Comment (merlimat) [inline pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:102]: we could add `hidden=true` here

Comment (RobertIndie) [reply pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java]: Thanks. Added.

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java [hunk 2]
```diff
@@ -164,19 +175,23 @@ public static void main(String[] args) throws Exception {
...
+        }
+
         ClientConfiguration bkConf = new ClientConfiguration();
         bkConf.setUseV2WireProtocol(true);
         bkConf.setAddEntryTimeout(30);
         bkConf.setReadEntryTimeout(30);
         bkConf.setThrottleValue(0);
         bkConf.setNumChannelsPerBookie(arguments.maxConnections);
-        bkConf.setZkServers(arguments.zookeeperServers);
+        bkConf.setMetadataServiceUri(arguments.metadataStoreUrl);
```
Comment (gaoran10) [inline pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:188]: It seems that only the param `arguments.metadataStoreUrl` is valid, right?

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java [hunk 3]
```diff
@@ -151,6 +156,12 @@ public static void main(String[] args) throws Exception {
             PerfClientUtils.exit(-1);
         }
 
+        if (arguments.metadataStoreUrl == null && arguments.zookeeperServers == null) {
```
Comment (gaoran10) [inline pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:159]: Does the param `arguments.zookeeperServers ` still be useful?

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java [hunk 4]
```diff
@@ -98,10 +98,15 @@
                 description = "Number of threads writing", validateWith = PositiveNumberParameterValidator.class)
         public int numThreads = 1;
 
-        @Parameter(names = { "-zk", "--zookeeperServers" },
-                description = "ZooKeeper connection string", required = true)
+        @Parameter(names = {"-zk", "--zookeeperServers"},
```
Comment (codelipenghui) [inline pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:101]: Add @Deprecated

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java (hunk 3 above)
Comment (RobertIndie) [reply pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:159]: Yes, we need to make it backward compatible.

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java (hunk 2 above)
Comment (RobertIndie) [reply pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:188]: Yes.

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java (hunk 4 above)
Comment (RobertIndie) [reply pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:102]: Thanks. Added.

**Review by Jason918 — APPROVED** [review]

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java [hunk 5]
```diff
@@ -98,10 +98,16 @@
...
-        @Parameter(names = { "-zk", "--zookeeperServers" },
-                description = "ZooKeeper connection string", required = true)
+        @Deprecated
+        @Parameter(names = {"-zk", "--zookeeperServers"},
+                description = "ZooKeeper connection string",
+                hidden = true)
         public String zookeeperServers;
 
+        @Parameter(names = {"-md",
+                "--metadata-store"}, description = "Metadata store service url. eg: zk:my-zk:2181")
```
Comment (Anonymitaet) [inline pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java:108]:
```suggestion
                "--metadata-store"}, description = "Metadata store service URL. For example: zk:my-zk:2181")
```

Please capitalize proper nouns correctly. Details: https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.je5thtkj42cb

**Review by merlimat — APPROVED** [review]

**Pull request merged**